src/map/unit.cpp

```diff
@@ -2910,6 +2910,21 @@ void unit_dataset(struct block_list *bl)
 	ud->canmove_tick   = gettick();
 }
 
+/**
+ * Returns the remaining max amount of skill units per object for a specific skill
+ * @param ud: Unit data
+ * @param skill_id: Skill to search for
+ * @param maxcount: Maximum amount of placeable units
+ */
+void unit_skillunit_maxcount(unit_data *ud, uint16 skill_id, int *maxcount) {
```

Comment:
```suggestion
void unit_skillunit_maxcount(unit_data& ud, uint16 skill_id, int& maxcount) {
```
Otherwise your code below doesnt work. Or you need to dereference the pointer. Right now you check if the pointer is > 0 and decrease the pointer with --, but not the value behind it.